parts/apicheck.pl

```diff
@@ -181,11 +183,26 @@
 static char  *VARarg2;
 static double VARarg3;
 
-#if defined(PERL_BCDVERSION) && (PERL_BCDVERSION < 0x5009005)
+#if defined(D_PPP_BCDVERSION) && (D_PPP_BCDVERSION < 0x5009005)
 /* needed to make PL_parser apicheck work */
 typedef void yy_parser;
 #endif
 
+/* Handle both 5.x.y and 7.x.y and up
+#ifndef PERL_MAJOR_VERSION
+#  define PERL_MAJOR_VERSION PERL_REVISION
```

Comment:
note that after Paul comment these macros are going to use a different naming
PERL_VERSION_MAJOR, ... MINOR, PATCH

Reply:
I hadn't realized the word order had changed

Reply:
yeap this is my fault and forgot to let you know